app/src/main/java/com/lagradost/quicknovel/MadaraInterface.kt

```diff
@@ -0,0 +1,317 @@
+import java.util.*
+import org.jsoup.Jsoup
+import org.jsoup.Connection
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+
+// nneed help for adding cookies and dns cache
+fun connect(url: String): Connection = Jsoup.connect(url).apply {
+    referrer("http://www.google.com")
+    userAgent(USER_AGENT)
+    header("Content-Language", "en-US")
+    header("Accept", "text/html")
+    header("Accept-Encoding", "gzip,deflate")
+    maxBodySize(0)
+    ignoreHttpErrors(true)
+}
+
+fun JConnect(url: String): Document? {
+    try {
+        val res = connect(url)
+            .timeout(20 * 1000)
+            .execute()
+        return if (res.statusCode() == 200) res.parse() else null
+    } catch (e: Exception) {
+        return null
+        /*
+        // maybe used for show dialog error
+        when (e) {
+            // - if the request URL is not a HTTP or HTTPS URL
+            // or is otherwise malformed
+            is MalformedURLException -> ""
+            // - if the response is not OK and HTTP response
+            // errors are not ignored
+            is HttpStatusException -> ""
+            // - if the response mime type is not supported
+            // and those errors are not ignored
+            is UnsupportedMimeTypeException ->
+            // - if the connection times out
+            is SocketTimeoutException -> ""
+            // - on IO error
+            is IOException -> ""
+            else -> ""
+        } */
+    }
+}
+
+fun String.toRate(multiply: Int): Int? {
+    return this
+        ?.toFloatOrNull()
+        ?.times(multiply)
+        ?.toInt() ?: 0
+}
+
+fun String.toVote(): Int? {
+    val k = this.contains("K", true)
+    val nodigit = "[^.0-9]".toRegex()
+    return this
+        .replace(nodigit, "")
+        .toFloatOrNull()
+        ?.times(if (k) 1000 else 1)
+        ?.toInt() ?: 0
+}
+
+fun String.clean(): String {
+    val reSpace = "[ ]+".toRegex()
+    val reTabN = "[\\n\\t\\r]+".toRegex()
+    return this
+        .replace(reTabN, "")
+        .replace(reSpace, " ")
+}
+
+fun String.toUrl(): Uri? = runCatching { Uri.parse(this) }.getOrNull()
+fun String.toUrlBuilder(): Uri.Builder? = toUrl()?.buildUpon()
+fun String.toUrlBuilderSafe(): Uri.Builder = toUrl()?.buildUpon()!!
+fun Uri.Builder.ifCase(case: Boolean, action: Uri.Builder.() -> Uri.Builder) = when {
+    case -> action(this)
+    else -> this
+}
+
+fun Uri.Builder.addPath(vararg path: String) =
+    path.fold(this) { builder, s ->
+        builder.appendPath(s)
+    }
+
+fun Uri.Builder.add(vararg query: Pair<String, Any>) =
+    query.fold(this) { builder, s ->
+        builder.appendQueryParameter(s.first, s.second.toString())
+    }
+
+fun Uri.Builder.add(key: String, value: Any): Uri.Builder =
+    appendQueryParameter(key, value.toString())
+
+abstract class MadaraInterface : MainAPI() {
+    open override val name = ""
+    open override val mainUrl = ""
+    open override val iconId = R.drawable.big_icon_boxnovel
```

Comment:
You should use the favicon of the site, using an existing icon will confuse a lot of users

Reply:
pleae someone give icon. I don't know how to do it 😁😁😁😁

Reply:
@KillerDogeEmpire please give it icon 😁😁😁

Reply:
@AhmadNaruto
I believe I have a icon from when you first made the providers. I can add an icon

Reply:
> @AhmadNaruto I believe I have a icon from when you first made the providers. I can add an icon

thanks 🙏😁

Reply:
@AhmadNaruto
sorry it has taken me sometime. I will add the Icons soon I have been busy with school